lib/src/pick_int.dart

```diff
@@ -10,38 +10,39 @@ extension NullableIntPick on Pick {
   /// Parses the picked value as [int]. Also tries to parse [String] as [int]
   /// via [int.tryParse]
   /// {@endtemplate}
-  int _parse() {
+  int _parse(bool roundDoubleToInt) {
     final value = required().value;
     if (value is int) {
       return value;
     }
+    if (value is num && roundDoubleToInt) {
+      return value.round();
+    }
     if (value is String) {
       final parsed = int.tryParse(value);
       if (parsed != null) {
         return parsed;
       }
     }
-    throw PickException(
-        'Type ${value.runtimeType} of $debugParsingExit can not be parsed as int');
+    throw PickException('Type ${value.runtimeType} of $debugParsingExit can not be parsed as int');
   }
 
   /// Returns the picked [value] as [int] or throws
   ///
   /// {@macro Pick.asInt}
-  int asIntOrThrow() {
-    withContext(requiredPickErrorHintKey,
-        'Use asIntOrNull() when the value may be null/absent at some point (int?).');
-    return _parse();
+  int asIntOrThrow({bool roundDoubleToInt = false}) {
+    withContext(requiredPickErrorHintKey, 'Use asIntOrNull() when the value may be null/absent at some point (int?).');
+    return _parse(roundDoubleToInt);
   }
 
   /// Returns the picked [value] as [int?] or returns `null` when the picked
   /// value is absent
   ///
   /// {@macro Pick.asInt}
-  int? asIntOrNull() {
+  int? asIntOrNull({bool roundDoubleToInt = false}) {
```

Comment:
If we offer `round()` we should also offer `toInt()`. That's why it was removed in the first place, because both are equally correct, it depends very much on the use case.

What do you think about adding `bool truncateDoubleToInt = false` as well?

Reply:
The method is already called `asIntOrNull`. `roundDoubleToInt` could therefore be shortened to `roundDouble`.

Reply:
done, but what should happen if a user sets both to true? throw an exception?

Reply:
yep, throw